Sources/Sentry/SentryDependencyContainer.m

```diff
@@ -76,18 +75,6 @@ - (SentryAppStateManager *)appStateManager
     }
 }
 
-- (SentryCrashWrapper *)crashWrapper
-{
-    if (_crashWrapper == nil) {
-        @synchronized(sentryDependencyContainerLock) {
-            if (_crashWrapper == nil) {
-                _crashWrapper = [SentryCrashWrapper sharedInstance];
-            }
-        }
-    }
-    return _crashWrapper;
-}
```

Comment:
h: I believe is better to keep access to crashWrapper here instead of using
```
[[[SentrySDK currentHub] getClient] crashWrapper]
```

Reply:
So we have only one place with all our dependencies.

Reply:
The point of this PR is to stop using the singleton because it's harder to know what pieces are being used in tests. Forcing all components to declare their dependencies in their constructors makes it easier to mock. This way the compiler can enforce this; with the singleton, you can easily forget to set a mock object unless you already know all the internals of the component you're about to test. This presents a high barrier to entry for any newcomer.

This has been an ongoing problem with dispatch queue wrappers and file managers, to name two that i know of.